### PR TITLE
Better merchant category names

### PR DESCRIPTION
**Pull request by 24c02**
now that yellow_pages can turn MCCs into reasonable names, let's have them everywhere!

### CODE REVIEW

File: app/models/card_grant.rb [hunk 1]
```diff
@@ -182,7 +182,7 @@ def allowed_categories
   end
 
   def allowed_category_names
-    allowed_categories.map(&:humanize)
+    allowed_categories.map { |category| YellowPages::Category.lookup(key: category).name || "#{category}*" }.uniq
```
Comment (garyhtou) [inline app/models/card_grant.rb]: out of curiosity, what's the purpose of the `*`?

File: app/views/hcb_codes/_stripe_card.html.erb [hunk 2]
```diff
@@ -295,7 +295,8 @@
           </p>
           <p>
             <strong>Category</strong>
-            <%= @hcb_code.stripe_merchant["category"]&.humanize %>
+            <%# we actually don't want humanized here – it's useful to see exactly what a merchant codes as in stripe_format for the purposes of setting up card grant allowlists %>
+            <%= @hcb_code.stripe_merchant["category"] %>
```
Comment (garyhtou) [inline app/views/hcb_codes/_stripe_card.html.erb]: Thoughts on doing "Fast food restaurants (`fast_food_restaurants`)"?

**Review by garyhtou — DISMISSED** [review]
tested. this functions well!

File: app/views/hcb_codes/_stripe_card.html.erb (hunk 2 above)
Comment (24c02) [reply app/views/hcb_codes/_stripe_card.html.erb]: yeah, i think i like that better!

File: app/models/card_grant.rb (hunk 1 above)
Comment (24c02) [reply app/models/card_grant.rb]: we don't enforce that card grant category restrictions are actual stripe categories, i'm using this as shorthand for "something might not be right here"

**Review by garyhtou — APPROVED** [review]

**Pull request merged**